reposerver/repository/repository.proto

```diff
@@ -254,6 +254,29 @@ message GitDirectoriesResponse {
     repeated string paths = 1;
 }
 
+message CompareRevisionsRequest {
+    github.com.argoproj.argo_cd.v2.pkg.apis.application.v1alpha1.Repository repo = 1;
+
+    string appLabelKey = 2;
+    string appName = 3;
+    string namespace = 4;
+    github.com.argoproj.argo_cd.v2.pkg.apis.application.v1alpha1.ApplicationSource applicationSource = 5;
+    string trackingMethod = 6;
+    map<string, github.com.argoproj.argo_cd.v2.pkg.apis.application.v1alpha1.RefTarget> refSources = 7;
+    string kubeVersion = 8;
+    repeated string apiVersions = 9;
+    bool hasMultipleSources = 10;
+
+    string syncedRevision = 11;
+    string revision = 12;
+    repeated string paths = 13;
+}
+
+message CompareRevisionsResponse {
+    bool changed = 1;
+    bool updated = 2;
```

Comment:
Is `updated` necessary here? It could be confusing to return that value to the client and it is not currently used.
Maybe the function should be called something like `SetLastRevisionForPaths` if the client does not do anything with the return object because the side effect of updating the cache to the last revision is the main purpose of the function.

Reply:
Agreed, I think the side effect should be explicit in the rpc name.

Reply:
Updated to `UpdateRevisionForPaths` and removed the `updated` field